The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11860250957/job/33055085105) [(plain)](https://github.com/rust-lang/rust/commit/7960a640439df5e77e24defbc5e39115a6b9c5f1/checks/33055085105/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
  PR_CI_JOB: 1
  READ_ONLY_SRC: 0
  IMAGE: x86_64-gnu-llvm-18
##[endgroup]
fatal: unknown commit 53fd98ca776cb875bc9e5514f56b52eb74f9e7a9
All commits in `HEAD` are present in `master`
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:c32c805632780b5c1de330e3f44561b336c2efe163bc0990acb392390157a8e1d9f855d75914a239aa40c49d77f4a837247d05d2f8d46f554b98e1f46712a3e3:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=beta', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
  Downloaded boml v0.3.1
   Compiling boml v0.3.1
   Compiling y v0.1.0 (/checkout/compiler/rustc_codegen_gcc/build_system)
    Finished `release` profile [optimized] target(s) in 3.80s
     Running `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-codegen/x86_64-unknown-linux-gnu/release/y test --use-system-gcc --use-backend gcc --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc --release --mini-tests --std-tests`
Using system GCC
[BUILD] example
[AOT] mini_core_hello_world
/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc/mini_core_hello_world
abc
---
Testing bufread-fill-buf.js ... OK
Testing deduplication.js ... OK
Testing enum-option.js ... OK
Testing exact-case.js ... FAILED
==> '{"path":"std::fs","name":"copy"}' was supposed to be before '{"crate":"std","ty":7,"name":"copy","path":"std::fs","descShard":{"crate":"std","shard":1,"start":2115,"len":2483,"promise":{}},"descIndex":2361,"exactPath":"std::fs","type":{"inputs":[{"id":-1,"ty":26,"path":null,"exactPath":null,"generics":[],"bindings":{}},{"id":-2,"ty":26,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"output":[{"id":14,"ty":8,"path":"std::io","exactPath":"std::io::error","generics":[{"id":59,"ty":1,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"bindings":{}}],"where_clause":[[{"id":468,"ty":10,"path":"std::convert","exactPath":"core::convert","generics":[{"id":652,"ty":5,"path":"std::path","exactPath":"std::path","generics":[],"bindings":{}}],"bindings":{}}],[{"id":468,"ty":10,"path":"std::convert","exactPath":"core::convert","generics":[{"id":652,"ty":5,"path":"std::path","exactPath":"std::path","generics":[],"bindings":{}}],"bindings":{}}]]},"id":55386,"word":"copy","normalizedName":"copy","bitIndex":10253,"implDisambiguator":null,"dist":0,"displayPath":"<span>std::</span><span>fs::</span>","fullPath":"std::fs::copy|7","href":"../std/fs/fn.copy.html","desc":"Copies the contents of one file to another. This function …"}'
Testing fn-forget.js ... OK
Testing from_u.js ... OK
Testing full-path-function.js ... OK
Testing iterator-type-signatures.js ... OK
---
Testing typed-query.js ... OK
Testing vec-new.js ... OK
Testing vec-type-signatures.js ... OK
Testing write.js ... FAILED
[ query `Write`]==> '{"path":"std::fmt","name":"write"}' was supposed to be before '{"crate":"std","ty":7,"name":"write","path":"std::fmt","descShard":{"crate":"std","shard":1,"start":2115,"len":2483,"promise":{}},"descIndex":2343,"exactPath":"core::fmt","type":{"inputs":[{"id":11,"ty":1,"path":null,"exactPath":null,"generics":[{"id":21,"ty":0,"path":null,"exactPath":null,"generics":[],"bindings":{}},{"id":147,"ty":10,"path":"std::fmt","exactPath":"core::fmt","generics":[],"bindings":{}}],"bindings":{}},{"id":133,"ty":5,"path":"std::fmt","exactPath":"core::fmt","generics":[],"bindings":{}}],"output":[{"id":14,"ty":6,"path":"std::result","exactPath":"core::result","generics":[{"id":4,"ty":1,"path":null,"exactPath":null,"generics":[],"bindings":{}},{"id":24,"ty":5,"path":"std::fmt","exactPath":"core::fmt","generics":[],"bindings":{}}],"bindings":{}}],"where_clause":[]},"id":55317,"word":"write","normalizedName":"write","bitIndex":10184,"implDisambiguator":null,"dist":0,"displayPath":"<span>std::</span><span>fmt::</span>","fullPath":"core::fmt::write|7","href":"../std/fmt/fn.write.html","desc":"Takes an output stream and an <code>Arguments</code> struct that can be …"}'
[ query `Write`]==> '{"path":"std::fs","name":"write"}' was supposed to be before '{"crate":"std","ty":7,"name":"write","path":"std::fs","descShard":{"crate":"std","shard":1,"start":2115,"len":2483,"promise":{}},"descIndex":2446,"exactPath":"std::fs","type":{"inputs":[{"id":-1,"ty":26,"path":null,"exactPath":null,"generics":[],"bindings":{}},{"id":-2,"ty":26,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"output":[{"id":14,"ty":8,"path":"std::io","exactPath":"std::io::error","generics":[{"id":4,"ty":1,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"bindings":{}}],"where_clause":[[{"id":468,"ty":10,"path":"std::convert","exactPath":"core::convert","generics":[{"id":652,"ty":5,"path":"std::path","exactPath":"std::path","generics":[],"bindings":{}}],"bindings":{}}],[{"id":468,"ty":10,"path":"std::convert","exactPath":"core::convert","generics":[{"id":1,"ty":1,"path":null,"exactPath":null,"generics":[{"id":15,"ty":1,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"bindings":{}}],"bindings":{}}]]},"id":55623,"word":"write","normalizedName":"write","bitIndex":10490,"implDisambiguator":null,"dist":0,"displayPath":"<span>std::</span><span>fs::</span>","fullPath":"std::fs::write|7","href":"../std/fs/fn.write.html","desc":"Writes a slice as the entire contents of a file."}'
[ query `Write`]==> '{"path":"std::ptr","name":"write"}' was supposed to be before '{"crate":"std","ty":7,"name":"write","path":"std::ptr","descShard":{"crate":"std","shard":3,"start":6858,"len":2362,"promise":{}},"descIndex":1106,"exactPath":"core::ptr","type":{"inputs":[{"id":-1,"ty":26,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"output":[{"id":4,"ty":1,"path":null,"exactPath":null,"generics":[],"bindings":{}}],"where_clause":[[]]},"id":63953,"word":"write","normalizedName":"write","bitIndex":18820,"implDisambiguator":null,"dist":0,"displayPath":"<span>std::</span><span>ptr::</span>","fullPath":"core::ptr::write|7","href":"../std/ptr/fn.write.html","desc":"Overwrites a memory location with the given value without …"}'
Command has failed. Rerun with -v to see more details.
  local time: Fri Nov 15 17:49:43 UTC 2024
  network time: Fri, 15 Nov 2024 17:49:43 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/ci/github-actions/jobs.yml
+++ b/src/ci/github-actions/jobs.yml
@@ -46,7 +46,7 @@ envs:
     RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
     MACOSX_DEPLOYMENT_TARGET: 10.12
     MACOSX_STD_DEPLOYMENT_TARGET: 10.12
-    SELECT_XCODE: /Applications/Xcode_14.3.1.app
+    SELECT_XCODE: /Applications/Xcode_15.2.app
     NO_LLVM_ASSERTIONS: 1
     NO_DEBUG_ASSERTIONS: 1
     NO_OVERFLOW_CHECKS: 1
@@ -282,7 +282,7 @@ auto:
       RUST_CONFIGURE_ARGS: --enable-full-tools --enable-sanitizers --enable-profiler --set rust.jemalloc --set rust.lto=thin --set rust.codegen-units=1
       RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
       MACOSX_DEPLOYMENT_TARGET: 10.12
-      SELECT_XCODE: /Applications/Xcode_14.3.1.app
+      SELECT_XCODE: /Applications/Xcode_15.2.app
       NO_LLVM_ASSERTIONS: 1
       NO_DEBUG_ASSERTIONS: 1
       NO_OVERFLOW_CHECKS: 1
@@ -298,7 +298,7 @@ auto:
       RUST_CONFIGURE_ARGS: --enable-sanitizers --enable-profiler --set rust.jemalloc --set target.aarch64-apple-ios-macabi.sanitizers=false --set target.x86_64-apple-ios-macabi.sanitizers=false
       RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
       MACOSX_DEPLOYMENT_TARGET: 10.12
-      SELECT_XCODE: /Applications/Xcode_14.3.1.app
+      SELECT_XCODE: /Applications/Xcode_15.2.app
       NO_LLVM_ASSERTIONS: 1
       NO_DEBUG_ASSERTIONS: 1
       NO_OVERFLOW_CHECKS: 1
@@ -327,7 +327,7 @@ auto:
         --set llvm.ninja=false
         --set rust.lto=thin
       RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
-      SELECT_XCODE: /Applications/Xcode_14.3.1.app
+      SELECT_XCODE: /Applications/Xcode_15.4.app
       USE_XCODE_CLANG: 1
       MACOSX_DEPLOYMENT_TARGET: 11.0
       MACOSX_STD_DEPLOYMENT_TARGET: 11.0
@@ -347,7 +347,7 @@ auto:
         --enable-profiler
         --set rust.jemalloc
       RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
-      SELECT_XCODE: /Applications/Xcode_14.3.1.app
+      SELECT_XCODE: /Applications/Xcode_15.4.app
       USE_XCODE_CLANG: 1
       MACOSX_DEPLOYMENT_TARGET: 11.0
       MACOSX_STD_DEPLOYMENT_TARGET: 11.0

--- a/src/librustdoc/html/static/js/search.js
+++ b/src/librustdoc/html/static/js/search.js
@@ -2098,6 +2098,7 @@ class DocSearch {
         const sortResults = async(results, isType, preferredCrate) => {
             const userQuery = parsedQuery.userQuery;
             const casedUserQuery = parsedQuery.original;
+            const isMixedCase = casedUserQuery !== userQuery;
             const result_list = [];
             for (const result of results.values()) {
                 result.item = this.searchIndex[result.id];
@@ -2109,10 +2110,12 @@ class DocSearch {
                 let a, b;
 
                 // sort by exact case-sensitive match
-                a = (aaa.item.name !== casedUserQuery);
-                b = (bbb.item.name !== casedUserQuery);
-                if (a !== b) {
-                    return a - b;
+                if (isMixedCase) {
+                    a = (aaa.item.name !== casedUserQuery);
+                    b = (bbb.item.name !== casedUserQuery);
+                    if (a !== b) {
+                        return a - b;
+                    }
                 }
 
                 // sort by exact match with regard to the last word (mismatch goes later)

--- a/tests/debuginfo/constant-ordering-prologue.rs
+++ b/tests/debuginfo/constant-ordering-prologue.rs
@@ -20,11 +20,11 @@
 // lldb-command:run
 
 // lldb-command:print a
-// lldb-check: = 19
+// lldb-check: 19
 // lldb-command:print b
-// lldb-check: = 20
+// lldb-check: 20
 // lldb-command:print c
-// lldb-check: = 21.5
+// lldb-check: 21.5
 
 fn binding(a: i64, b: u64, c: f64) {
     let x = 0;

--- a/tests/rustdoc-js-std/write.js
+++ b/tests/rustdoc-js-std/write.js
@@ -1,0 +1,24 @@
+const EXPECTED = [
+    {
+        'query': 'write',
+        'others': [
+            { 'path': 'std::fmt', 'name': 'write' },
+            { 'path': 'std::fs', 'name': 'write' },
+            { 'path': 'std::ptr', 'name': 'write' },
+            { 'path': 'std::fmt', 'name': 'Write' },
+            { 'path': 'std::io', 'name': 'Write' },
+            { 'path': 'std::hash::Hasher', 'name': 'write' },
+        ],
+    },
+    {
+        'query': 'Write',
+        'others': [
+            { 'path': 'std::fmt', 'name': 'Write' },
+            { 'path': 'std::io', 'name': 'Write' },
+            { 'path': 'std::fmt', 'name': 'write' },
+            { 'path': 'std::fs', 'name': 'write' },
+            { 'path': 'std::ptr', 'name': 'write' },
+            { 'path': 'std::hash::Hasher', 'name': 'write' },
+        ],
+    },
+];

--- a/tests/rustdoc-js/case.js
+++ b/tests/rustdoc-js/case.js
@@ -1,0 +1,17 @@
+const EXPECTED = [
+    {
+        'query': 'Foo',
+        'others': [
+            { 'path': 'case', 'name': 'Foo', 'desc': 'Docs for Foo' },
+            { 'path': 'case', 'name': 'foo', 'desc': 'Docs for foo' },
+        ],
+    },
+    {
+        'query': 'foo',
+        'others': [
+            // https://github.com/rust-lang/rust/issues/133017
+            { 'path': 'case', 'name': 'Foo', 'desc': 'Docs for Foo' },
+            { 'path': 'case', 'name': 'foo', 'desc': 'Docs for foo' },
+        ],
+    },
+];

--- a/tests/rustdoc-js/case.rs
+++ b/tests/rustdoc-js/case.rs
@@ -1,0 +1,7 @@
+#![allow(nonstandard_style)]
+
+/// Docs for Foo
+pub struct Foo;
+
+/// Docs for foo
+pub struct foo;

